boards: riscv: hifive1: Add GPIO-controlled LEDs

soburi opened the pull request:
Add GPIO-controlled LEDs and alias them as led0, led1, led2. Rename the PWM controlled LEDs to pwmled0, pwmled2, pwmled2.

Add empty prj.conf to `samples/shields/x_nucleo_53l0a1` to fix CI failure.


Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi commented:
Hmm, It seem need to fix renode first....

PiotrZierhoffer commented:
@soburi we're looking into adding the required features to the HiFive1 GPIO controller model in Renode

soburi commented:
@PiotrZierhoffer 

Thank you for suggestion.

```
2022-12-18T12:32:20.3471398Z 12:31:25.8245 [WARNING] gpioInputs: Unhandled write to offset 0x24, value 0x80000.
2022-12-18T12:32:20.3471798Z 12:31:25.8247 [WARNING] gpioInputs: Unhandled write to offset 0x2C, value 0x80000.
2022-12-18T12:32:20.3472221Z 12:31:25.8247 [WARNING] gpioInputs: Unhandled write to offset 0x34, value 0x80000.
```

https://github.com/renode/renode-infrastructure/blob/03baf69b3c61b56e8f1c2ca04661a928b8592f4e/src/Emulator/Peripherals/Peripherals/GPIOPort/SiFive_GPIO.cs

At a glance, It already implements `RiseInterruptPending` and I think it can solve with also implementing Fall/High/Low InterruptPendings to suppress the warnings.
Saying within this PR scope, The CI failed with timeout by copious log messages.

(I have yet to gain experience in renode, So I need to learn how to fix renode.)

mateusz-holenko commented:
Hi @soburi,
extended version of the `SiFive_GPIO` model supporting more types of interrupts is now available upstream:  https://github.com/renode/renode/commit/32e2fa06054373f12d627375f1c7cc9878520eaf.
With this new version we were able to successfully pass the `drivers/gpio/gpio_api_1pin/peripheral.gpio.1pin` test on HiFive1 in Renode.

Next steps on our side are to prepare a new Renode release and make Zephyr use it.

soburi commented:
@mateusz-holenko 

Great job! 
I'm looking forward to pass this test with the next release Renode.
Thanks a lot!

fkokosinski commented:
Hi @soburi, the newest Renode release should now be available in the CI. Could you rebase this PR onto the `main` branch?

soburi committed:
boards: riscv: hifive1: Add GPIO-controlled LEDs

Add GPIO-controlled LEDs and alias them as led0, led1, led2.
Rename the PWM controlled LEDs to pwmled0, pwmled2, pwmled2.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi commented:
@fkokosinski 
Rebased and one add more modification to make pass CI.

jfischer-no commented on boards/riscv/hifive1/hifive1.dts:
```diff
@@ -25,16 +29,32 @@
+	pwmleds {
+		compatible = "pwm-leds";
+		pwmled0: pwmled_0 {
+			pwms = <&pwm1 1 PWM_MSEC(20)>;
```
Should it be `pwms = <&pwm1 1 PWM_MSEC(20) PWM_POLARITY_INVERTED>;`?

soburi replied:
Currently hifive1's PWM only takes channel and period parameters. I'll leave it as is for this PR.

jfischer-no commented on samples/shields/x_nucleo_53l0a1/prj.conf:
```diff
@@ -0,0 +1 @@
+# nothing here
```
You can drop this commit, fixed by https://github.com/zephyrproject-rtos/zephyr/pull/55973

soburi replied:
Rebased and drop it.
Thank you for notifiy.
